File: backend/src/models/feedModel.js
/**
 * Feed/Feeding record data structure for Firestore
 * Collection: feed/{feedId}
 */
const FeedModel = {
  animalCategory: "", // e.g., Cow, Goat, Sheep, Pig, Chicken
  userId: "",         // Who recorded this feeding
  type: "",        // e.g., Hay, Grain, Silage, Grass, Pellets
  quantity: 0,     // Amount in kg or liters
  unit: "kg",      // Unit of measurement: kg | liters | bales
  cost: 0,         // Cost of this feeding session
  notes: "",       // Optional notes
  date: "",        // ISO timestamp of when feeding occurred
  createdAt: "",   // Record creation time
};

module.exports = FeedModel;

File: frontend/grazetrack_app/lib/features/feeding/models/feed_model.dart
class FeedModel {
  final String id;
  final String animalCategory;
  final String userId;
  final String type;
  final double quantity;
  final String unit;
  final double cost;
  final String notes;
  final String date;

  const FeedModel({
    required this.id,
    required this.animalCategory,
    required this.userId,
    required this.type,
    required this.quantity,
    this.unit = 'kg',
    required this.cost,
    this.notes = '',
    required this.date,
  });

  factory FeedModel.fromJson(Map<String, dynamic> json) => FeedModel(
        id: json['id'] ?? '',
        animalCategory: json['animalCategory'] ?? json['animalId'] ?? '',
        userId: json['userId'] ?? '',
        type: json['type'] ?? '',
        quantity: (json['quantity'] ?? 0).toDouble(),
        unit: json['unit'] ?? 'kg',
        cost: (json['cost'] ?? 0).toDouble(),
        notes: json['notes'] ?? '',
        date: json['date'] ?? '',
      );

  Map<String, dynamic> toJson() => {
        'animalCategory': animalCategory,
        'type': type,
        'quantity': quantity,
        'unit': unit,
        'cost': cost,
        'notes': notes,
        'date': date,
      };
}
